tokio/src/runtime/context/blocking.rs

```diff
@@ -32,7 +32,7 @@ pub(crate) fn try_enter_blocking_region() -> Option<BlockingRegionGuard> {
 
 /// Disallows blocking in the current runtime context until the guard is dropped.
 pub(crate) fn disallow_block_in_place() -> DisallowBlockInPlaceGuard {
-    let reset = CONTEXT.with(|c| {
+    let reset = CONTEXT.try_with(|c| {
```

Comment:
It might be worth aligning here too?
https://github.com/tokio-rs/tokio/blob/ab8d7b82a1252b41dc072f641befb6d2afcb3373/tokio/src/runtime/context/blocking.rs#L109

I think drop still gets panic if `CONTEXT.try_with` returns Err when constructing a `DisallowBlockInPlaceGuard`.

Reply:
If the first `try_with` fails, then the guard stores `false` so the `if` makes us skip that `CONTEXT.with` call.